public/app/features/explore/state/actions.ts

```diff
@@ -313,10 +306,36 @@ export function loadDatasource(exploreId: ExploreId, instance: DataSourceApi): T
     }
 
     if (datasourceError) {
-      dispatch(loadDatasourceFailureAction({ exploreId, error: datasourceError }));
-      return Promise.reject(`${datasourceName} loading failed`);
```

Comment:
Do we need the Promise.reject()?

Reply:
possibly, as the caller does a try catch, and if error skips execute queries. 
https://github.com/grafana/grafana/blob/3395b7411635774d418ec67e1bb657d66daf9acf/public/app/features/explore/state/actions.ts#L121

Reply:
Good catch, I'll fix.

Reply:
Changed to check if state has `datasourceError` before running any queries